Fix issue #1390, ODataUriParser can't pass function call with all omi…

xuzhg opened the pull request:
…tted optional parameters

<!-- markdownlint-disable MD002 MD041 -->

### Issues

*This pull request fixes issue #1390.*

### Description

If a function call is with all omitted optional parameter, OData Uri parser can't parse it.
See the detail in the issue.

For example:
`NS.GetMinSalary` is a bound function with two optional parameters, one is "minSalary", the other is "maxSalary".

OData Uri parser can parse:
```C#
~/Customers/NS.GetMinSalary(minSalary=1,maxSalary=2)
~/Customers/NS.GetMinSalary(minSalary=1)
```
But, fails to the following:

```C#
~/Customers/NS.GetMinSalary()
```
This PR is fix that.

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

mikepizzo commented on test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Parsers/FunctionOverloadResolverTests.cs:
```diff
@@ -288,7 +287,47 @@ public void OverloadBoundFunctionsSuccessfullyResolved()
+
+            IEdmOperation function;
+
+            // Resolve overload function without parameter. Be noted, it picks the 
```
"...one with no optional parameters."

mikepizzo commented on test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Parsers/FunctionOverloadResolverTests.cs:
```diff
@@ -288,7 +287,47 @@ public void OverloadBoundFunctionsSuccessfullyResolved()
+            FunctionOverloadResolver.ResolveOperationFromList("NS.Function", parameters, int32TypeReference.Definition, model, out function, DefaultUriResolver).Should().BeTrue();
+            function.Should().BeSameAs(functionWithoutParameter);
 
+            // Resolve overload function with one required and three optional parameters (one omitted).
```
>required and three optional parameters (one omitted) [](start = 50, length = 52)

"...bound and two optional parameters."

mikepizzo commented on test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Parsers/FunctionOverloadResolverTests.cs:
```diff
@@ -288,7 +287,47 @@ public void OverloadBoundFunctionsSuccessfullyResolved()
+            FunctionOverloadResolver.ResolveOperationFromList("NS.Function", parameters, int32TypeReference.Definition, model, out function, DefaultUriResolver).Should().BeTrue();
+            function.Should().BeSameAs(functionWithAllOptionalParameters);
+
+            // Resolve overload function with one required and three optional parameters (one omitted).
```
>required and three optional parameters (one omitted) [](start = 50, length = 52)

"...bound and two optional parameters."

mikepizzo commented:
minor comments on the test descriptions; otherwise looks good.

xuzhg commented:
@Mike, thanks for the quick review. I addressed the comments. Thanks.

---
In reply to: [461620523](https://github.com/OData/odata.net/pull/1399#issuecomment-461620523) [](ancestors = 461620523)
